cmake/modules/FindSYMENGINE.cmake

```diff
@@ -118,7 +118,19 @@ STRING(REGEX REPLACE
   )
 REMOVE_DUPLICATES(_symengine_include_dirs)
 
-SET(_symengine_libraries ${SYMENGINE_LIBRARIES})
+#
+# The SYMENGINE_LIBRARIES variable configured by SymEngine only lists the
+# libraries, but does not set their paths. So we configure this outselves.
```

Comment:
Are you sure about this? `SYMENGINE_LIBRARIES` is supposed to include a mix of targets (with the location set in the target) and absolute paths.

Reply:
Yeah, I am sure:
```
#
# Include the SymEngine:
#
FIND_PACKAGE(SymEngine ${SYMENGINE_MINIMUM_REQUIRED_VERSION}
  CONFIG QUIET
  HINTS ${SYMENGINE_DIR}
  PATH_SUFFIXES lib/cmake/symengine
  NO_SYSTEM_ENVIRONMENT_PATH
  )
  
MESSAGE(STATUS "LISTING THE SYMENGINE_LIBRARIES: ${SYMENGINE_LIBRARIES}")
```
prints
```
-- LISTING THE SYMENGINE_LIBRARIES: symengine
```
with https://github.com/symengine/symengine/commit/e66c30c60da65d4a0a058fd992c3c2f6b20be3d6

But I understand that this is probably not what is intended to happen on this line:
https://github.com/symengine/symengine/blob/master/cmake/SymEngineConfig.cmake.in#L101
Missing `" "` maybe?

Reply:
[Here's the `SymEngineConfig.cmake`](https://gist.github.com/jppelteret/eb31b8de6c8410a051394add4fb9110f) that's generated for me from  symengine/symengine@e66c30c. The following lines are suspiciously empty:
```
set(SYMENGINE_LLVM_LIBRARIES )
set(SYMENGINE_LLVM_COMPONENTS )

if (NOT "<non-empty boost path>" STREQUAL "")
    ...
    set(SYMENGINE_LIBRARIES ${SYMENGINE_LIBRARIES} )
endif()

```

Reply:
Try doing,
```
foreach (LIB ${SYMENGINE_LIBRARIES})
    if ("${LIB}" STREQUAL symengine)
        get_property(SYMENGINE_LOC TARGET ${LIB} PROPERTY LOCATION)
        message("${SYMENGINE_LOC}")
    endif()
endforeach()
```

and you'll see that `symengine` is a target even though it prints as `symengine`